File: c#Tests/tests/msTesting/UnitTest1.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace msTesting
{
    [TestClass]
    public class UnitTest1
    {
        Calculator calc = new Calculator();
        [TestMethod]
        public void TestMethod1()
        {
            //Arrange
            int a = 10;
            int b = 20;
            int expectedResult = 30;
            //Act
            int result = calc.add(a, b);
            //Assert
            Assert.AreEqual(expectedResult, result);
        }
        [TestMethod]
        [DataRow(10, 30, 40)]
        [DataRow(15, 10, 25)]
        [DataRow(9, 1, 10)]
        public void TestMethod2(int x, int y, int expected)
        {
            //Act
            int result = calc.add(x, y);
            //Assert
            Assert.AreEqual(result, expected);
        }
        [TestMethod]
        [DynamicData(nameof(AdditionData))]
        public static IEnumerable<object[]> AdditionData()
        {
            yield return new object[] { 10, 20, 30 };
            yield return new object[] { 50, 25, 75 };
            yield return new object[] { 15, 10, 25 };
        }
        public void TestMethod3(int x, int y, int expected)
        {
            //Act
            int result = calc.add(x, y);
            //Assert
            Assert.AreEqual(result, expected);
        }
    }
}

File: cpp tests/ZSK_CPP_Tests/UnitTest1/UnitTest1.cpp
#include "pch.h"
#include "CppUnitTest.h"
#include "../ZSK_CPP_Tests/Algorithms.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace UnitTest1
{
	TEST_CLASS(UnitTest1)
	{
	public:
		TEST_METHOD(TestMethod1)
		{
			//Arrange
			int n = 5;
			int expected = 120;
			//Act
			int actual = Algorithms::factorial(n);
			//Assert
			Assert::AreEqual(expected, actual);
		}
		TEST_METHOD(TestMethod2)
		{
			//Arrange
			int n = 5;
			int expected = 15;
			int arr[] = { 1,2,3,4,5 };
			//Act
			int actual = Algorithms::arraySum(arr, n);
			//Assert
			Assert::AreEqual(expected, actual);
		}
		TEST_METHOD(TestMethod3)
		{
			//Arrange
			long long base = 2;
			long long exp = 10;
			long long expected = 1024;
			//Act
			long long actual = Algorithms::fastPow(base, exp);
			//Assert
			Assert::AreEqual(expected, actual);
		}
		TEST_METHOD(TestMethod4)
		{
			//Arrange
			int u = 12;
			int v = 18;
			int expected = 6;
			//Act
			int actual = Algorithms::gcd_recursive(u, v);
			//Assert
			Assert::AreEqual(expected, actual);
		}	
	};
}
